### PR TITLE
Add pull_request_target for fork PRs

### PR DESCRIPTION
**Pull request by miraai**
**Proposed changes**:
- Update workflows to be able to run on PRs from forks if label `security-review:done` is assigned to the PR
- Simplify docker build/push step
- Add checkout only from request HEAD commit if is fork

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by rasa-jmac — COMMENTED** [review]
This looks good to me from a first pass.

I think we can simplify the build/push logic (lines 664 onwards) because they were originally split out from a "build + push" step into "build" and "push if not a fork" in the first attempt to solve this problem. If this all looks like it works we can refactor it back into how it was before: https://github.com/RasaHQ/rasa/pull/11590/files

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -617,55 +675,36 @@ jobs:
...
         run: |
           export IMAGE_TAG=${{ steps.check_image.outputs.base_image_hash }}
-          docker buildx bake -f docker/docker-bake.hcl base --push
-
-      - name: Build Docker mitie base image 🛠
-        if: steps.check_image.outputs.base_mitie_exists == 'false'
-        run: |
-          export IMAGE_TAG=${{ steps.check_image.outputs.base_mitie_image_hash }}
-          docker buildx bake -f docker/docker-bake.hcl base-mitie
+          docker buildx bake --set *.platform=linux/amd64 -f docker/docker-bake.hcl base --push
```
Comment (rasa-jmac) [inline .github/workflows/continous-integration.yml:684]:
```suggestion
          docker buildx bake -f docker/docker-bake.hcl base --push
```

File: .github/workflows/continous-integration.yml [hunk 2]
```diff
@@ -617,55 +675,36 @@ jobs:
...
+          docker buildx bake --set *.platform=linux/arm64 -f docker/docker-bake.hcl base --push
```
Comment (rasa-jmac) [inline .github/workflows/continous-integration.yml:685]:
```suggestion

```

**Review by rasa-jmac — CHANGES_REQUESTED** [review]
This looks good to me, I like how you've cleaned up the conditions.

I realise now the PR I linked you to as a reference had different architecture builds for the Docker images but that was a test that didn't go anywhere unrelated to the PR. For each of the Docker build steps, we can just include one `docker buildx --push` command instead of the two for each architecture, eg:

```
docker buildx bake -f docker/docker-bake.hcl base
```

instead of the two architectures:
```
 docker buildx bake --set *.platform=linux/amd64 -f docker/docker-bake.hcl base --push
 docker buildx bake --set *.platform=linux/arm64 -f docker/docker-bake.hcl base --push
```

Does that make sense?

**Comment by rasa-jmac** [comment]
Also, I wonder if we should change the label from `status:fork` to something more explicit that a dev would be less likely to add by accident like `security-review:done` or `dangerous:acknowledged`

**Comment by miraai** [comment]
> Also, I wonder if we should change the label it needs from `status:fork` to something more explicit that a dev would be less likely to add by accident like `security-review:done` or `privileged:yes`

Oh yeah, definitely. I used `status:fork` more like a placeholder until we agree on a label name.

**Review by rasa-jmac — COMMENTED** [review]
Looks good except one condition where it looks like you've swapped the logic

File: .github/workflows/continous-integration.yml [hunk 3]
```diff
@@ -674,7 +713,7 @@ jobs:
       # Example output:
       # IMAGE_TAG=1234
       - name: Set environment variables - pull_request
-        if: github.event_name == 'pull_request' && env.IS_TAG_BUILD == 'false'
+        if: env.IS_TAG_BUILD && (github.event_name == 'pull_request' || env.IS_PULL_REQUEST_TARGET_FORK == 'true')
```
[the remark this answers is not in the document]
Comment (miraai) [reply .github/workflows/continous-integration.yml:716]: Good catch, that wasn't on purpose, probably missed it when i was rewriting it. Thanks!

**Review by rasa-jmac — APPROVED** [review]
LGTM!

**Review by rasa-jmac — APPROVED** [review]
LGTM!

**Pull request merged**